### PR TITLE
Multiple test fixes and improved logging for fragile tests

### PR DESCRIPTION
**Pull request by adityapatwardhan**
# PR Summary

* Read-Host - set the expect file to be executable
* DotNetAPI - lookup a type instead of calling an API that does not work on all platforms
* ExecutionPolicy - Fix test and improve logging for the test
* GetCommand - Fix test
* Markdown cmdlet - Fix tests on platforms that do not support VT100.
* BasicEngine - Add more logging for fragile test. This fails intermittently. 
* TaskBasedAsyncPowerShellAPI - Add a sleep for waiting for the pipeline to start. Also added more logging.
* HelpSystem.Online - Fix the expected environment variable as AzDevOps makes are variable all caps.

## PR Context

Fix multiple test failures found while running release automation tests on multiple platforms.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

### CODE REVIEW

File: test/powershell/Modules/Microsoft.PowerShell.Security/ExecutionPolicy.Tests.ps1 [hunk 1]
```diff
@@ -667,7 +667,7 @@ ZoneId=$FileType
                 $testData += @(
                     @{
                         module = $PSHomeUntrustedModule
-                        error = $null
+                        error = $error
```
Comment (JamesWTruher) [inline test/powershell/Modules/Microsoft.PowerShell.Security/ExecutionPolicy.Tests.ps1]: not sure about this - is this just whatever is there?

File: test/powershell/Modules/Microsoft.PowerShell.Utility/MarkdownCmdlets.Tests.ps1 [hunk 2]
```diff
@@ -27,6 +27,10 @@ Describe 'ConvertFrom-Markdown tests' -Tags 'CI' {
             [bool] $VT100Support
             )
 
+            # Force VT100Support to be false is the host does not support it.
```
Comment (JamesWTruher) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/MarkdownCmdlets.Tests.ps1]: typo `is` -> `if`?

File: test/powershell/Modules/Microsoft.PowerShell.Utility/MarkdownCmdlets.Tests.ps1 [hunk 3]
```diff
@@ -125,7 +129,11 @@ Describe 'ConvertFrom-Markdown tests' -Tags 'CI' {
         BeforeAll {
             $mdFile = New-Item -Path $TestDrive/input.md -Value "Some **test string** to write in a file" -Force
             $mdLiteralPath = New-Item -Path $TestDrive/LiteralPath.md -Value "Some **test string** to write in a file" -Force
-            $expectedStringFromFile = "Some $esc[1mtest string$esc[0m to write in a file`n`n"
+            $expectedStringFromFile = if ($host.UI.SupportsVirtualTerminal) {
```
Comment (JamesWTruher) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/MarkdownCmdlets.Tests.ps1]: should `$host.UI.SupportsVirtualTerminal` be `$VT100Support`?

File: test/powershell/Modules/Microsoft.PowerShell.Utility/MarkdownCmdlets.Tests.ps1 [hunk 4]
```diff
@@ -281,6 +289,7 @@ bool function()`n{`n}
 
     Context "ConvertFrom-Markdown empty or null content tests" {
         BeforeAll {
+            $vt100Supported = $host.UI.SupportsVirtualTerminal
```
Comment (JamesWTruher) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/MarkdownCmdlets.Tests.ps1]: why is this logic different from above?

**Review by JamesWTruher — COMMENTED** [review]
just nits and a typo. Should probably fix the typo

File: test/powershell/Modules/Microsoft.PowerShell.Security/ExecutionPolicy.Tests.ps1 (hunk 1 above)
Comment (adityapatwardhan) [reply test/powershell/Modules/Microsoft.PowerShell.Security/ExecutionPolicy.Tests.ps1]: We should have been expecting an error defined at line # 657. I will change the variable name to make it clearer.

File: test/powershell/Modules/Microsoft.PowerShell.Utility/MarkdownCmdlets.Tests.ps1 (hunk 2 above)
Comment (adityapatwardhan) [reply test/powershell/Modules/Microsoft.PowerShell.Utility/MarkdownCmdlets.Tests.ps1]: Fixed

File: test/powershell/Modules/Microsoft.PowerShell.Utility/MarkdownCmdlets.Tests.ps1 (hunk 3 above)
Comment (adityapatwardhan) [reply test/powershell/Modules/Microsoft.PowerShell.Utility/MarkdownCmdlets.Tests.ps1]: Moved the variable to Describe BeforeAll

File: test/powershell/Modules/Microsoft.PowerShell.Utility/MarkdownCmdlets.Tests.ps1 (hunk 4 above)
Comment (adityapatwardhan) [reply test/powershell/Modules/Microsoft.PowerShell.Utility/MarkdownCmdlets.Tests.ps1]: Merged the logic now.

**Comment by PoshChan** [comment]
@adityapatwardhan, your last commit had 1 failures in `PowerShell-CI-windows`
Validate ExecutionPolicy cmdlets in PowerShell.Validate that 'Unrestricted' execution policy works on OneCore powershell.Test 'Unrestricted' execution policy. Importing D:\a\1\s\src\powershell-win-core\bin\Release\netcoreapp2.1\win7-x64\publish\Modules\Loc
```powershell
Expected an exception, with FullyQualifiedErrorId 'UnauthorizedAccess,Microsoft.PowerShell.Commands.ImportModuleCommand' to be thrown, but no exception was thrown.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Security\ExecutionPolicy.Tests.ps1: line 688
688:                     $testScript | Should -Throw -ErrorId $expectedError
```

**Comment by adityapatwardhan** [comment]
@JamesWTruher I have addressed your comments. Please have another look.

File: test/powershell/Host/ConsoleHost.Tests.ps1 [hunk 5]
```diff
@@ -780,8 +780,9 @@ public enum ShowWindowCommands : int
             Start-Sleep -Milliseconds 100
             $showCmd = ([Test.User32]::GetPlacement($ps.MainWindowHandle)).showCmd
         }
-        $showCmd | Should -BeExactly $WindowStyle
+
         $ps | Stop-Process -Force
```
Comment (TravisEz13) [inline test/powershell/Host/ConsoleHost.Tests.ps1]: This line should be in a finally?

File: test/powershell/Modules/Microsoft.PowerShell.Security/ExecutionPolicy.Tests.ps1 [hunk 6]
```diff
@@ -678,15 +679,23 @@ ZoneId=$FileType
...
-                    $testScript | Should -Throw -ErrorId $error
+                    Set-ItResult -Pending -Because "Test is unreliable"
+                }
+
+                $execPolicy = Get-ExecutionPolicy -List | Out-String
+
+                $testScript = {Import-Module -Name $module -Force -ErrorAction Stop}
+                if($expectedError)
+                {
+                    $testScript | Should -Throw -ErrorId $expectedError
```
Comment (TravisEz13) [inline test/powershell/Modules/Microsoft.PowerShell.Security/ExecutionPolicy.Tests.ps1]: NIT: add because statement here too

**Review by TravisEz13 — APPROVED** [review]

File: test/powershell/Host/ConsoleHost.Tests.ps1 (hunk 5 above)
Comment (adityapatwardhan) [reply test/powershell/Host/ConsoleHost.Tests.ps1]: Fixed

File: test/powershell/Modules/Microsoft.PowerShell.Security/ExecutionPolicy.Tests.ps1 (hunk 6 above)
Comment (adityapatwardhan) [reply test/powershell/Modules/Microsoft.PowerShell.Security/ExecutionPolicy.Tests.ps1]: Fixed

**Review by TravisEz13 — APPROVED** [review]

**Comment by TravisEz13** [comment]
Does this need to be back ported?

**Pull request merged**